Name: Improved CIMG Photo Naming
Version: 1.4-1
Author: Jason Robitaille
Description: Will output images from the camera in the format: CIMG_YYYY-MM-DD-hhmmss.jpg  (MIT license)

--- .orig/usr/palm/applications/com.palm.app.camera/javascripts/camera-control.js
+++ /usr/palm/applications/com.palm.app.camera/javascripts/camera-control.js
@@ -62,6 +62,17 @@
 			// A (default) filename was passed in - presumably we've been sublaunched and the caller
 			// knows where they want to save the file.
 			cam.imageDestination = this.scene.defaultFilename;
+		} else {
+			var date = new Date();
+			var YYYY = date.getFullYear();
+			var MM = (1+date.getMonth()); if (MM<10) { MM = "0"+MM; }
+			var DD = (0+date.getDate()); if (DD<10) { DD = "0"+DD; }
+			var hh = date.getHours(); if (hh < 10) { hh = "0"+hh; }
+			var mm = date.getMinutes(); if (mm < 10) { mm = "0"+mm; }
+			var ss = date.getSeconds(); if (ss < 10) { ss = "0"+ss; }
+			var fileBase = "_" + YYYY + "-" +  MM + "-" + DD + "-" + hh + mm + ss;
+			cam.imageDestination = "/media/internal/DCIM/100PALM/" + 
+					"CIMG" + fileBase + ".jpg";
 		}
 
 		// Call the capture state machine.		
